feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radius_proxy.c

```diff
@@ -0,0 +1,231 @@
+		[RADIUS_PROXY_SERVER_TYPE] = { .name = "type", BLOBMSG_TYPE_STRING },
+		[RADIUS_PROXY_SERVER_SECRET] = { .name = "secret", BLOBMSG_TYPE_STRING },
+		[RADIUS_PROXY_SERVER_STATUS] = { .name = "statusServer", BLOBMSG_TYPE_BOOL },
+		[RADIUS_PROXY_SERVER_CERT_NAME_CHECK] = { .name = "certificateNameCheck", BLOBMSG_TYPE_BOOL },
+		[RADIUS_PROXY_PORT] = { .name = "port", BLOBMSG_TYPE_STRING },
+		[RADIUS_PROXY_REALM_NAME] = { .name = "name", BLOBMSG_TYPE_STRING },
+		[RADIUS_PROXY_REALM_SERVER] = { .name = "server", BLOBMSG_TYPE_ARRAY },
+};
+
+const struct uci_blob_param_list radius_proxy_param = {
+	.n_params = __RADIUS_PROXY_MAX,
+	.params = radsec_proxy_policy,
+};
+
+size_t file_write(void *ptr, size_t size, size_t nmemb, FILE *stream) {
+	size_t written = fwrite(ptr, size, nmemb, stream);
+	return written;
+}
+
+static bool radsec_download_cert(char *cert_name, char *dir_name, char *cert_url)
+{
+	CURL *curl;
+	FILE *fp;
+	CURLcode res;
+	char path[200];
+	char name[32];
+	char dir[32];
+
+	strcpy(name, cert_name);
+	strcpy(dir, dir_name);
+	sprintf(path, "/usr/radsec/certs/%s/%s", dir, name);
+
+	curl = curl_easy_init();
+	if (curl)
+	{
+		fp = fopen(path,"wb");
+
+		if (fp == NULL)
+		{
+			curl_easy_cleanup(curl);
+			return false;
+		}
+
+		if (cert_url == NULL)
+		{
+			curl_easy_cleanup(curl);
+			fclose(fp);
+			return false;
+		}
+
+		curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L);
+		curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L);
+		curl_easy_setopt(curl, CURLOPT_HEADER, 0L);
+		curl_easy_setopt(curl, CURLOPT_URL, cert_url);
+		curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, file_write);
+		curl_easy_setopt(curl, CURLOPT_WRITEDATA, fp);
+		res = curl_easy_perform(curl);
+		curl_easy_cleanup(curl);
+		fclose(fp);
+		return res;
+	}
+
+	return true;
+}
+
+static bool raidus_proxy_config_set( struct schema_Radius_Proxy_Config *old,
+	struct schema_Radius_Proxy_Config *conf )
+{
+	int i=0;
+	char path[200];
+
+	blob_buf_init(&uci_buf, 0);
+	n = blobmsg_open_array(&uci_buf,"ListenUDP");
+	blobmsg_add_string(&uci_buf, NULL, "127.0.0.1:1812");
+	blobmsg_close_array(&uci_buf, n);
+	blob_to_uci_section(uci, "radsecproxy", NULL, "options", uci_buf.head, &radius_proxy_param, NULL);
+
+	blob_buf_init(&uci_buf, 0);
+	blobmsg_add_string(&uci_buf, "name", "localhost");
+	blobmsg_add_string(&uci_buf, "type", "udp");
+	blobmsg_add_string(&uci_buf, "secret", "secret");
+	blob_to_uci_section(uci, "radsecproxy", NULL, "client", uci_buf.head, &radius_proxy_param, NULL);
+
+	blob_buf_init(&uci_buf, 0);
+	if (strlen(conf->server))
+	{
+		blobmsg_add_string(&uci_buf, "name", conf->server);
+		blobmsg_add_string(&uci_buf, "type", "tls");
+		blobmsg_add_string(&uci_buf, "secret", "radsec");
+		blobmsg_add_bool(&uci_buf, "statusServer", 1);
+		blobmsg_add_bool(&uci_buf, "certificateNameCheck", 0);
+		blob_to_uci_section(uci, "radsecproxy", NULL, "server", uci_buf.head, &radius_proxy_param, NULL);
+	}
+
+	for (i = 0; i < conf->realm_len; i++)
```

Comment:
What does this do?  I think it creates multiple sections of type "realm" with a distinct "name" option conf->realm[i] and the same "server" value conf->server

Reply:
This code was rewritten to generate a separate block per realm.